python/lsst/daf/persistence/butler.py

```diff
@@ -1245,17 +1245,23 @@ def datasetExists(self, datasetType, dataId={}, **rest):
         datasetType = self._resolveDatasetTypeAlias(datasetType)
         dataId = DataId(dataId)
         dataId.update(**rest)
+        location = self._locate(datasetType, dataId, write=False)
+        if location is None:
+            return False
 
-        location = None
-        for repoData in self._repos.inputs():
-            if not dataId.tag or len(dataId.tag.intersection(repoData.tags)) > 0:
-                location = repoData.repo.map(datasetType, dataId)
-                if location and location.repository.exists(location):
-                    break
+        exists = None
+        if isinstance(location, ButlerComposite):
```

Comment:
I find ambiguity in the name "ButlerComposite" and its relation to location in the comparison, especially the first part "Butler", in the local context - a comment would help.